test/dist_optimizer_test.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import unittest
+
+from dist_utils import INIT_METHOD_TEMPLATE, dist_init
+import torch
+import torch.distributed.autograd as dist_autograd
+import torch.distributed.optimizer.dist_optimizer as dist_optimizer
+import torch.distributed.rpc as rpc
+
+class MyModule:
+    def __init__(self):
+        torch.manual_seed(0)
+        self.w = torch.rand((3, 3), requires_grad=True)
+
+    def forward(self, t1):
+        return torch.mm(self.w, t1)
+
+    def get_w(self):
+        return self.w
+
+
+class FailingOptimizer(dist_optimizer.FunctionalOptimizer):
+    def step(self, gradients):
+        raise ValueError('Error running optimizer.')
+
+
+def _call_meth(meth, obj_rref, *args, **kwargs):
+    return meth(obj_rref.local_value().wait(), *args, **kwargs)
+
+
+def remote_meth(meth, obj_rref, *args, **kwargs):
+    """
+    Call rpc.remote on a method in a remote object.
+
+    Args:
+        meth: the method (for example, Class.method)
+        obj_rref (RRef): remote reference to the object
+        args: positional arguments to pass to the method
+        kwargs: keyword arguments to pass to the method
+
+    Returns a RRef to the remote method call result.
+    """
+    return rpc.remote(
+        obj_rref.owner(),
+        _call_meth,
+        args=[meth, obj_rref] + list(args),
+        kwargs=kwargs
+    )
+
+
+def rpc_async_meth(meth, obj_rref, *args, **kwargs):
+    """
+    Call rpc.remote on a method in a remote object.
+
+    Args:
+        meth: the method (for example, Class.method)
+        obj_rref (RRef): remote reference to the object
+        args: positional arguments to pass to the method
+        kwargs: keyword arguments to pass to the method
+
+    Returns a Future to the method call result.
+    """
+    return rpc.rpc_async(
+        obj_rref.owner(),
+        _call_meth,
+        args=[meth, obj_rref] + list(args),
+        kwargs=kwargs
+    )
+
+
+@unittest.skipIf(
+    not torch._six.PY3, "Pytorch distributed autograd package " "does not support python2"
```

Comment:
two unnecessary double quotes, or maybe you were intended to break them into shorter lines.